test/unit/processing/siti-agri/build-invoice-lines.test.js

```diff
@@ -42,97 +43,92 @@ describe('Build invoice lines', () => {
   })
 
   test('should not overwrite agreement number for invoice lines to provided contract number if not CS', async () => {
-    const invoiceLinesParse = buildInvoiceLines(sfi23.schemeId, invoiceLines, 'C123')
-    expect(invoiceLinesParse[0].agreementNumber).toBe(invoiceLines[0].agreementNumber)
+    const invoiceLinesParse = buildInvoiceLines(paymentRequest)
+    expect(invoiceLinesParse[0].agreementNumber).toBe(paymentRequest.invoiceLines[0].agreementNumber)
   })
 
-  test('should overwrite agreement number for invoice lines to provided contract number if CS', async () => {
-    const invoiceLinesParse = buildInvoiceLines(cs.schemeId, invoiceLines, 'C123')
-    expect(invoiceLinesParse).toMatchObject([
-      {
-        schemeCode: 'SITIELM',
-        accountCode: 'ABC123',
-        fundCode: 'ABC12',
-        agreementNumber: 'C123',
-        description: 'G00 - Gross value of claim',
-        value: 100,
-        convergence: true,
-        deliveryBody: 'RP00',
-        marketingYear: 2023
-      }
-    ])
+  test('should overwrite agreement number for invoice lines to header level contract number if CS', async () => {
+    paymentRequest.schemeId = cs.schemeId
+    const invoiceLinesParse = buildInvoiceLines(paymentRequest)
+    expect(invoiceLinesParse[0].agreementNumber).toBe(paymentRequest.contractNumber)
+  })
+
+  test('should overwrite delivery body for invoice lines to header level delivery body if CS', async () => {
+    paymentRequest.schemeId = cs.schemeId
+    const invoiceLinesParse = buildInvoiceLines(paymentRequest)
+    expect(invoiceLinesParse[0].deliveryBody).toBe(paymentRequest.deliveryBody)
```

Comment:
Could we add another test that tests an existing line value is overwritten?

Just to protect the behaviour from future changes to the code.